### PR TITLE
Remove or ignore remaining pylint issues and fail on any pylint problem

### PR DESCRIPTION
**Pull request by collisdigital**
### What is the context of this PR?
- Either fixed or set a disable flag for all remaining pylint issues
- Changed `run_lint.sh` to fail on ANY pylint issue (warning, error etc.)
- Changed `run_lint.sh` to stop doing full reports, they are noise and output in pretty colours

### How to review 
- Run `./scripts/run_lint.sh` and check it passes
- Review code changes or pylint disabled options added.

### CODE REVIEW

File: app/forms/fields.py [hunk 1]
```diff
@@ -164,17 +165,17 @@ def get_integer_field(answer, label, guidance, error_messages):
     if answer['type'] == 'Integer':
         validate_with = mandatory_or_optional + [
             IntegerCheck(answer_errors['NOT_INTEGER']),
-            NumberRange(max=9999999999, messages=answer_errors),
+            NumberRange(maximum=9999999999, messages=answer_errors),
```
Comment (collisdigital) [inline app/forms/fields.py:168]: min/max clash with the Python built-in with the same name

File: app/forms/fields.py [hunk 2]
```diff
@@ -142,6 +142,7 @@ class CustomIntegerField(IntegerField):
     """
     def __init__(self, **kwargs):
         super(CustomIntegerField, self).__init__(**kwargs)
+        self.data = None
```
Comment (collisdigital) [inline app/forms/fields.py:145]: Pylint complained that self.data is used later in this class but not declared in `__init__`

File: app/data_model/database.py [hunk 3]
```diff
@@ -79,7 +79,8 @@ def _create_session_and_engine_with_retry():
     for i in range(settings.EQ_SERVER_SIDE_STORAGE_DATABASE_SETUP_RETRY_COUNT):
         try:
             return _create_session_and_engine()
-        except Exception as e:
+        except Exception as e:  # pylint: disable=broad-except
```
Comment (collisdigital) [inline app/data_model/database.py:82]: Hard to predict what can go wrong when trying to setup the database; so this retries x times before giving up and throwing the last exception caught.

File: app/templating/summary_context.py [hunk 4]
```diff
@@ -18,12 +18,12 @@ def build_summary_rendering_context(schema_json, answer_store, metadata):
...
             if block['id'] in [location.block_id for location in path] and block['type'] == 'Questionnaire':
-                    link = url_for('questionnaire.get_block',
-                                   eq_id=metadata['eq_id'],
-                                   form_type=metadata['form_type'],
-                                   collection_id=metadata['collection_exercise_sid'],
-                                   group_id=group['id'],
-                                   group_instance=0,
-                                   block_id=block['id'])
-                    sections.extend([Section(section, answer_store.map(), link) for section in block['sections']])
+                link = url_for('questionnaire.get_block',
```
Comment (collisdigital) [inline app/templating/summary_context.py:21]: indent issue resolved

File: app/validation/validators.py [hunk 5]
```diff
@@ -51,26 +51,26 @@ class NumberRange(object):
     This will work with any comparable number type, such as floats and
     decimals, not just integers.
 
-    :param min:
+    :param minimum:
```
Comment (collisdigital) [inline app/validation/validators.py:54]: min/max clash with the Python built-in with the same name

File: scripts/run_lint.sh [hunk 6]
```diff
@@ -21,14 +21,12 @@ function display_result {
 flake8 --max-complexity 10 --count
 display_result $? 1 "Flake 8 code style check"
 
-pylint --rcfile=.pylintrc -j 0 ./app ./tests
+pylint --reports=n --output-format=colorized --rcfile=.pylintrc -j 0 ./app ./tests
```
Comment (collisdigital) [inline scripts/run_lint.sh:24]: we only get errors or warnings output to console now; not the full code-tree and tabulated results etc. errors and warnings are coloured coded

File: tests/app/parser/test_metadata_parser.py [hunk 7]
```diff
@@ -247,20 +247,20 @@ def test_is_valid_fails_invalid_return_by(self):
...
-             "form_type": "a",
-             "collection_exercise_sid": "test-sid",
-             "eq_id": "2",
-             "period_id": "3",
-             "period_str": "2016-01-01",
-             "ref_p_start_date": "2016-02-02",
-             "ru_ref": "2016-04-04",
-             "ru_name": "Apple",
-             "return_by": "2016-07-07"
+            "user_id": "1",
```
Comment (collisdigital) [inline tests/app/parser/test_metadata_parser.py:250]: indent issue

File: tests/app/parser/test_metadata_parser.py [hunk 8]
```diff
@@ -247,20 +247,20 @@ def test_is_valid_fails_invalid_return_by(self):
...
+            "eq_id": "2",
+            "period_id": "3",
+            "period_str": "2016-01-01",
+            "ref_p_start_date": "2016-02-02",
+            "ru_ref": "2016-04-04",
+            "ru_name": "Apple",
+            "return_by": "2016-07-07"
         }
-        valid, field = is_valid_metadata(jwt)
+        valid, _ = is_valid_metadata(jwt)
```
Comment (collisdigital) [inline tests/app/parser/test_metadata_parser.py:261]: unused field

File: scripts/run_lint.sh (hunk 6 above)
Comment (benjefferies) [reply scripts/run_lint.sh:24]: Much better output

**Review by benjefferies — CHANGES_REQUESTED** [review]
Looks good. Only thing is the linting doesn't run on travis anymore (my fault)

**Comment by collisdigital** [comment]
Going to rebase and squash this now

**Review by benjefferies — APPROVED** [review]

**Comment by collisdigital** [comment]
There's a code coverage issue on this PR where an indent level has been changed and the line wasn't covered by a test; suggest we ignore it as unrelated to the change being made.

**Comment by ajmaddaford** [comment]
`app/parser/parser_utils.py` isn't used, so may be worth removing that and the associated tests and  see if the coverage increases enough to hit the target?

**Comment by collisdigital** [comment]
@ajmaddaford will remove `app/parser/parser_utils.py`

**Comment by collisdigital** [comment]
@ajmaddaford removed but didn't change coverage %

**Review by ajmaddaford — APPROVED** [review]

**Pull request merged**